vault/identity_store_aliases.go

```diff
@@ -256,6 +247,51 @@ func (i *IdentityStore) handleAliasUpdateCommon(req *logical.Request, d *framewo
 		return nil, err
 	}
 
+	var existingEntity *identity.Entity
+	var lockKeys []string
+	if !newAlias {
```

Comment:
I think this part should be after the locks are grabbed. The MemDB stuff happens in a transaction but it's still possible for that transaction to say, for instance, that there is no existingEntity, then have such an entity be created after the locking.

Reply:
Unfortunately, we don't know the id of the entity to grab the lock for.  This is what I alluded to as still a race but since we have to look it up, I'm not sure the alternative.

Reply:
One way to alleviate this issue would be to add locks for alias IDs.  I'll went down that path at one point but pulled it out since it was only really used here.  I think that would fully take care of this issue and will do that.